router/core/router.go

```diff
@@ -224,9 +227,32 @@ func NewRouter(opts ...Option) (*Router, error) {
 // updateServer starts a new Server. It swaps the active Server with a new Server instance when the config has changed.
 // This method is safe for concurrent use. When the router can't be swapped due to an error the old server kept running.
 func (r *Router) updateServer(ctx context.Context, cfg *nodev1.RouterConfig) error {
+	overrides := make(map[string]string)
+
+	for _, sg := range cfg.Subgraphs {
```

Comment:
Put this logic in a method for better readability e.g. `configureSubgraphOverwrites`.